core/src/main/java/com/linecorp/armeria/common/stream/EventLoopStreamMessage.java

```diff
@@ -126,18 +126,20 @@ protected EventExecutor defaultSubscriberExecutor() {
     }
 
     @Override
-    void subscribe(SubscriptionImpl subscription) {
-        final Subscriber<?> subscriber = subscription.subscriber();
+    SubscriptionImpl subscribe(SubscriptionImpl subscription) {
         if (!subscribedUpdater.compareAndSet(this, 0, 1)) {
-            eventLoop.execute(() -> failLateSubscriber(this.subscription, subscriber));
-            return;
+            final SubscriptionImpl oldSubscription = this.subscription;
+            assert oldSubscription != null;
```

Comment:
This can be `false` because at this point the `subscription` couldn't be set.
It's no worry to fix it using `subscriptionUpdater` like we do in other classes (not `subscribedUpdater`), but I just want to make it sure that it's worth because I couldn't find any usage of this.
@trustin, @anuraaga, do we need this `EventLoopStreamMessage`?

Reply:
I think we can go ahead and remove the class now, other performance improvements make it less meaningful from looking at benchmraks.

Reply:
Ok, got it. Will remove it then. Thanks! :bow: